combat-trainer.lic

```diff
@@ -3407,7 +3411,7 @@ class AttackProcess
   end
 
   def dance(game_state)
-    if game_state.npcs.empty?
+    if game_state.npcs.empty? || !game_state.is_engaging_allowed?
```

Comment:
Do dance actions break Innocence? If not, should this check be moved to line 3420?

Reply:
Yes, the `engage` on line 3416 will break Innocence. Any `face` or `engage` will. (But for whatever reason, `support` won't, and implicitly facing the next target with a `target` `bob` etc won't...)

But dance commands like `weave` and `circle` require you to be at pole range or closer, which is incompatible with Innocence which keeps enemies entirely disengaged if it works.